### ISSUE TITLE
MultiThreadedExecutor produces random errors

### ISSUE THREAD

**Issue by d01010101** [ros2-java/ros2_java#59]
This works fine:

        RCLJava.rclJavaInit();
        Executor mx = new SingleThreadedExecutor();
        mx.addNode(new SubscriberMemberFunction());
        mx.addNode(new PublisherMemberFunction());
        mx.spin();
        RCLJava.shutdown();

But if SingleThreadedExecutor is replaced by MultiThreadedExecutor, one of the following messages is produced:

########################### message 1
    0 [main] INFO org.ros2.rcljava.common.JNIUtils  - Loading implementation: org_ros2_rcljava_rcl_java__jni
    7 [main] INFO org.ros2.rcljava.RCLJava  - Using RMW implementation: rmw_fastrtps_cpp
    9 [main] INFO org.ros2.rcljava.common.JNIUtils  - Loading implementation: org_ros2_rcljava_executors__base_executor__jni
    19 [main] INFO org.ros2.rcljava.common.JNIUtils  - Loading implementation: org_ros2_rcljava_node__node_impl__jni
    89 [main] INFO org.ros2.rcljava.common.JNIUtils  - Loading typesupport: std_msgs_msg__string__jni__rosidl_typesupport_c
    91 [main] INFO org.ros2.rcljava.common.JNIUtils  - Loading implementation: org_ros2_rcljava_subscription__subscription_impl__jni
    98 [main] INFO org.ros2.rcljava.common.JNIUtils  - Loading implementation: org_ros2_rcljava_publisher__publisher_impl__jni
    101 [main] INFO org.ros2.rcljava.common.JNIUtils  - Loading implementation: org_ros2_rcljava_timer__wall_timer_impl__jni
    #
    # A fatal error has been detected by the Java Runtime Environment:
    #
    #  SIGSEGV (0xb) at pc=0x00007f78159b1a88, pid=15280, tid=0x00007f77e67fc700
    #
    # JRE version: Java(TM) SE Runtime Environment (8.0_181-b13) (build 1.8.0_181-b13)
    # Java VM: Java HotSpot(TM) 64-Bit Server VM (25.181-b13 mixed mode linux-amd64 compressed oops)
    # Problematic frame:
    # C  [librmw_fastrtps_cpp.so+0x5ba88]  rmw_wait+0x187
    #
    # Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
    #
    # An error report file with more information is saved as:
    # /home/art/NetBeansProjects/VericsROS2/hs_err_pid15280.log
    #
    # If you would like to submit a bug report, please visit:
    #   http://bugreport.java.com/bugreport/crash.jsp
    # The crash happened outside the Java Virtual Machine in native code.
    # See problematic frame for where to report the bug.
    #

########################### message 2
    0 [main] INFO org.ros2.rcljava.common.JNIUtils  - Loading implementation: org_ros2_rcljava_rcl_java__jni
    7 [main] INFO org.ros2.rcljava.RCLJava  - Using RMW implementation: rmw_fastrtps_cpp
    8 [main] INFO org.ros2.rcljava.common.JNIUtils  - Loading implementation: org_ros2_rcljava_executors__base_executor__jni
    17 [main] INFO org.ros2.rcljava.common.JNIUtils  - Loading implementation: org_ros2_rcljava_node__node_impl__jni
    76 [main] INFO org.ros2.rcljava.common.JNIUtils  - Loading typesupport: std_msgs_msg__string__jni__rosidl_typesupport_c
    78 [main] INFO org.ros2.rcljava.common.JNIUtils  - Loading implementation: org_ros2_rcljava_subscription__subscription_impl__jni
    84 [main] INFO org.ros2.rcljava.common.JNIUtils  - Loading implementation: org_ros2_rcljava_publisher__publisher_impl__jni
    86 [main] INFO org.ros2.rcljava.common.JNIUtils  - Loading implementation: org_ros2_rcljava_timer__wall_timer_impl__jni
    java:         
/home/art/ros2_java_ws/src/ros2_java/ros2_java/rcljava/src/main/cpp/org_ros2_rcljava_timer_WallTimerImpl.cpp:40: jboolean Java_org_ros2_rcljava_timer_WallTimerImpl_nativeIsReady(JNIEnv*, jclass, jlong): 
    Assertion `wall_timer_handle != 0' failed.

#################################
java version "1.8.0_181"
Java(TM) SE Runtime Environment (build 1.8.0_181-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.181-b13, mixed mode)

Using built-in specs.
COLLECT_GCC=g++
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/7/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 7.3.0-16ubuntu3' --with-bugurl=file:///usr/share/doc/gcc-7/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++ --prefix=/usr --with-gcc-major-version-only --with-as=/usr/bin/x86_64-linux-gnu-as --with-ld=/usr/bin/x86_64-linux-gnu-ld --program-suffix=-7 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --with-sysroot=/ --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-libmpx --enable-plugin --enable-default-pie --with-system-zlib --with-target-system-zlib --enable-objc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 7.3.0 (Ubuntu 7.3.0-16ubuntu3) 

I use a recent snapshot.

**Comment by jacobperron** [comment]
Since it's been a while since this issue was opened and we've made a lot of changes since then, I'm going to close this ticket.

Please try the `main` branch with ROS Galactic and open a new ticket if the issue persists.
